src/CrawlerProperties.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace Spatie\Crawler;
+
+use Spatie\Robots\RobotsTxt;
+use Psr\Http\Message\UriInterface;
+use Spatie\Browsershot\Browsershot;
+use Spatie\Crawler\CrawlQueue\CrawlQueue;
+
+trait CrawlerProperties
+{
+    public function setConcurrency(int $concurrency): self
```

Comment:
Isn't a `self` return type problematic when extending a class? I know you'd rather not have this issue, but we should be consistent when using `protected`...

Reply:
You're right, let's discuss this further in person though.